src/test/resources/com/puppycrawl/tools/checkstyle/xpath/abstractrootnode/InputAbstractRootNode.java

```diff
@@ -0,0 +1,18 @@
+/*
+com.puppycrawl.tools.checkstyle.filters.SuppressionXpathFilter
+file = (file)InputAbstractRootNode.xml
+optional = (default)false
+
+com.puppycrawl.tools.checkstyle.checks.coding.IllegalTokenTextCheck
+format = [^a-zA-z0-9]*
+ignoreCase = (default)false
+message = (default)
+tokens = CHAR_LITERAL
+
+*/
+package com.puppycrawl.tools.checkstyle.xpath.abstractrootnode;
+
+public class InputAbstractRootNode {
```

Comment:
SuppressionXpathFilter vs InputAbstractRootNode is red flag.

please move this code to SuppressionXpathFilterTest.
fact that we have some abstract class does not mean that we need to test is as abstract. We test all from user perspective, user knows only SuppressionXpathFilter and amount of abstract classes or classes used by composite that are under it does not matter.

if you keep all tests at level of SuppressionXpathFilter we can refactor all easily and we will never need to change a test.

Reply:
Done